tee-worker/core-primitives/types/src/lib.rs

```diff
@@ -91,11 +90,35 @@ impl Encode for OpaqueCall {
 	}
 }
 
-// Note in the pallet teerex this is a struct. But for the codec this does not matter.
+// Litentry: re-defined due to orphan rule - it needs to be kept identical to that in teerex-primitives
```

Comment:
Can that be enforced by a compiler check and/or unit test? e.g. encoding from one struct and decoding from the other should yield the same information? :thinking:

Reply:
That's a good idea, thanks @grumpygreenguy  - I'll look into it but maybe a macro declaration should be enough